packages/cozy-sharing/src/components/RecipientsAvatars.spec.jsx

```diff
@@ -0,0 +1,104 @@
+import React from 'react'
+import { createMockClient } from 'cozy-client'
+import { render } from '@testing-library/react'
+
+import { RecipientsAvatars, MAX_DISPLAYED_RECIPIENTS } from './Recipient'
+import AppLike from '../../test/AppLike'
+
+const mockRecipients = new Array(MAX_DISPLAYED_RECIPIENTS - 1)
+  .fill(
+    {
+      status: 'owner',
+      public_name: 'cozy'
+    },
+    0,
+    1
+  )
+  .fill(
+    {
+      status: 'pending',
+      name: 'Mitch Young'
+    },
+    1
+  )
+
+const mockMoreRecipientsThanMaxDisplayed = [
+  ...mockRecipients,
+  {
+    status: 'mail-not-send',
+    name: 'Lyn Webster'
+  },
+  {
+    status: 'ready',
+    name: 'Richelle Young'
+  },
+  {
+    status: 'ready',
+    name: 'John Connor'
+  }
+]
+
+describe('RecipientsAvatars', () => {
+  const client = createMockClient({})
+
+  const setup = ({
+    recipients = mockRecipients,
+    link = false,
+    isOwner = true,
+    onClick = () => jest.fn(),
+    ...rest
+  }) => {
+    return render(
+      <AppLike client={client}>
+        <RecipientsAvatars
+          recipients={recipients}
+          onClick={onClick}
+          isOwner={isOwner}
+          link={link}
+          {...rest}
+        />
+      </AppLike>
+    )
+  }
+
+  it('should render link icon if a link is generated', () => {
+    const { getByTestId } = setup({
+      link: true
+    })
+
+    expect(getByTestId('recipientsAvatars-link')).toBeTruthy()
+  })
+
+  it('should not render link icon if a link is not generated', () => {
+    const { queryByTestId } = setup({})
+
+    expect(queryByTestId('recipientsAvatars-link')).toBeNull()
+  })
+
+  it('should hide me as owner by default', () => {
+    const { queryByTestId } = setup({})
+
+    expect(queryByTestId('recipientsAvatars-avatar-owner')).toBeNull()
+  })
+
+  it('should show me as owner if required', () => {
+    const { getByTestId } = setup({
+      showMeAsOwner: true
+    })
+
+    expect(getByTestId('recipientsAvatars-avatar-owner')).toBeTruthy()
+  })
+
+  it('should show a +X icon with the correct number if there is more avatars than expected', () => {
```

Comment:
On devrait regarder si le link est toujours présent aussi tant qu'à faire 👍

Reply:
Hmmm c'est des choses indépendantes, et c'est couvert par le test `should render link icon if a link is generated`. Mais ok je rajoute un test `should show both +X and link icon if necessary` qui teste le cas `showMeAsOwner: true, link: true`